experiments/src/viewer.ts

```diff
@@ -0,0 +1,21 @@
+import { getViewBuilder } from "@polkadot-api/substrate-codegen"
+import { readFile } from "node:fs/promises"
+import { v14 } from "@polkadot-api/substrate-bindings"
+
+const metadataBytes = await readFile("./ksm.scale")
+const metadata = v14.dec(metadataBytes)
+
+const { callDecoder } = getViewBuilder(metadata)
+const result = callDecoder(
+  "0x17002b0f01590100004901415050524f56455f52464328303030352c39636261626661383035393864323933353833306330396331386530613065346564383232376238633866373434663166346134316438353937626236643434290101000000",
+)
+
+console.log(
```

Comment:
I'm not sure if this is redundant or work as designed.
The `shape` seems to be repeated inside value

<img width="1029" alt="image" src="https://github.com/paritytech/polkadot-api/assets/1209171/279bae03-3722-4a05-9407-11f63bc4defc">

Reply:
This was done intentionally. The reasoning was that while analyzing an Enum value -unless the redundant outer "shape" is there- the consumer would only be able to know about the "shape" of the selected variant. This would be unfortunate, because the visualizer may want to display the other variants available. 

However, now that I think about it, this would only be a problem if the root value being decoded was an enum. Otherwise, the consumer could obtain the shape of the other variants from its parent.

Therefore, a better solution would be to only report the "redundant" shape alongside with the root `Decoded` 🤔.

Yeah! I think that I'm going to change this. Thanks @kratico !